src/compiler/utilities.ts

```diff
@@ -11,6 +11,36 @@ namespace ts {
         isTypeReferenceDirective?: boolean;
     }
 
+    export function getNameOfDeclaration(declaration: Declaration): DeclarationName {

```

Comment:
Can't we just use the type `Declaration | BinaryExpression` here? If not I would rather we have a `DeclarationLike` alias for that rather than renaming `Declaration` to `RealDeclaration`.